src/io/fileIO.cu

```diff
@@ -0,0 +1,651 @@
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "fileIO.h"
+#include "H5Cpp.h"
+using namespace H5;
+
+/* HDF5 structure:
+ * # data_dir/output.h5
+ * /ENERGY
+ * /WFC
+ * /WFC/CONST
+ * /WFC/CONST/i
+ * /WFC/EV
+ * /WFC/EV/i
+ * /V
+ * /V/i
+ * /K
+ * /K/i
+ * /VORTEX/EDGES/i
+ * /A
+ * /A/AX
+ * /A/AY
+ * /A/AZ
+ * /DOMAIN
+ * /DOMAIN/X
+ * /DOMAIN/Y
+ * /DOMAIN/Z
+ *
+ * Where "/" is the root,
+ * Uppercase entries are groups,
+ * Lowercase entries are datasets,
+ * and "i" means the dataset at the i-th iteration
+ *
+*/
+
+#include "fileIO.h"
+
+namespace FileIO{
+
+    // Output file
+    H5File *output;
+
+    // Groups
```

Comment:
Curious if these are freed anywhere? I have seen that some of the pointer types are freed later, but I may be missing it for the rest of these. I wonder if a run through valgrind to check for leaks would be a good idea.

Also wondering if it would make sense to use unique_ptr types instead so that we do not need to manually manage memory safety. Thoughts?

Reply:
Yeah, the groups are missing from the deletion block.
I'll add them for clarity, but we are really only need to destroy the file, to ensure that it is closed.
Since these are never overwritten, they should hang around until program termination and be automatically freed, but I agree that it is better to err on the side of caution.

With regards to `unique_ptr`s, it would be great to use them, but I am using pointers here because they are used everywhere else.
Their types should remain consistent regardless of where they are.
If we do want to use `unique_ptr`s, we should use them everywhere.

That sounds like a bit of an undertaking, but doable.
I'll defer to @leios on making a call here

Reply:
Rewriting to use unique_ptr everywhere would be a bit of an undertaking. Adding the free's to the destroy method is fine by me. I'll have a more detailed look at the rest when I get a chance, but seems all good to me. Nice work.

Reply:
Just to clarify, the `delete` deallocates the pointer and triggers the destructor of the hdf5 objects, which in turn deallocate and close themselves. We don't need to explicitly call `free`